src/openforms/registrations/contrib/objects_api/tests/test_json_schema.py

```diff
@@ -0,0 +1,319 @@
+        ]
+
+        self.assertListEqual(
+            paths_list[1:],
+            [
+                (["complaintDescription"], {"type": "string"}),
+                (["measuredDecibels"], {"type": "array", "items": {"type": "number"}}),
+                (
+                    ["complainant"],
+                    {
+                        "properties": {
+                            "first.name": {"type": "string"},
+                            "last.name": {"type": "string"},
+                        },
+                        "type": "object",
+                    },
+                ),
+                (["complainant", "first.name"], {"type": "string"}),
+                (["complainant", "last.name"], {"type": "string"}),
+            ],
+        )
+
+    def test_iter_json_schema_refs(self):
+        paths_list = [
+            (path.segments, schema)
+            for path, schema in iter_json_schema_paths(JSON_SCHEMA_REFS)
+        ]
+
+        self.assertListEqual(
+            paths_list[1:],
+            [
+                (
+                    ["complainant"],
+                    {
+                        "properties": {
+                            "first.name": {"type": "string"},
+                            "last.name": {"type": "string"},
+                        },
+                        "type": "object",
+                    },
+                ),
+                (["complainant", "first.name"], {"type": "string"}),
+                (["complainant", "last.name"], {"type": "string"}),
+                (
+                    ["noisyAddress"],
+                    {
+                        "properties": {
+                            "street": {"type": "string"},
+                        },
+                        "type": "object",
+                    },
+                ),
+                (["noisyAddress", "street"], {"type": "string"}),
+            ],
+        )
+
+    def test_iter_json_schema_nested_refs(self):
+        paths_list = [
+            (path.segments, schema)
+            for path, schema in iter_json_schema_paths(JSON_SCHEMA_NESTED_REFS)
+        ]
+
+        self.assertListEqual(
+            paths_list[1:],
+            [
+                (
+                    ["complainant"],
+                    {
+                        "properties": {"residence": {"$ref": "#/definitions/address"}},
+                        "type": "object",
+                    },
+                ),
+                (
+                    ["complainant", "residence"],
+                    {"properties": {"street": {"type": "string"}}, "type": "object"},
+                ),
+                (["complainant", "residence", "street"], {"type": "string"}),
+            ],
+        )
+
+    def test_iter_json_schema_unknown_ref(self):
+        self.assertRaises(
+            Unresolvable,
+            lambda: list(
+                iter_json_schema_paths(JSON_SCHEMA_UNKOWN_REF, fail_fast=True)
+            ),
+        )
```

Comment:
we use the context manager form usually:

```py
with self.assertRaises(Unresolvable):
    list(iter_json_schema_paths(JSON_SCHEMA_UNKOWN_REF, fail_fast=True))
```

it's a bit more readable than the lambda variant.